src/python/pants/engine/objects.py

```diff
@@ -70,15 +70,19 @@ def is_serializable(obj):
 
     :rtype: bool
     """
-    return isinstance(obj, Serializable) or (not inspect.isclass(obj) and hasattr(obj, '_asdict'))
+    if inspect.isclass(obj):
+      return Serializable.is_serializable_type(obj)
+    return isinstance(obj, Serializable) or hasattr(obj, '_asdict')
```

Comment:
Fix to issue discussed [here](https://github.com/pantsbuild/pants/pull/6272/files/4c74e713c3cd6bb535b4cb6d1ce95485228ffe62#diff-52577b31050779942e23bcd9485633b0). Sometimes the passed value isn't a class, and in Py3 it complains about this whereas in Py2 it silently works.